packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -2038,9 +2084,14 @@ class AnnotatedRegionLayer<T> extends ContainerLayer {
   /// by the [size] and [offset] if provided.
   ///
   /// The [value] provided cannot be null.
-  AnnotatedRegionLayer(this.value, {this.size, Offset offset})
-      : offset = offset ?? Offset.zero,
-        assert(value != null);
+  AnnotatedRegionLayer(
+    this.value, {
+    this.size,
+    Offset offset,
+    this.opaque = false,
+  }) : assert(value != null),
+       assert(opaque != null),
+       offset = offset ?? Offset.zero;
```

Comment:
do we intentionally accept `null` as a value for `offset`?

Reply:
(as opposed to `Offset.zero` being the default value)

Reply:
When forwarding values (from subclass constructors, for instance), it often happens that just passing on a parameter that is missing converts it to a null.  It means that the same default needs to be applied in a lot of places,  instead of just in one, so this code is just guarding against that, and supplying the one default. And without this, if the default changes in one place, we need to remember to change it everywhere, which is error-prone.

Reply:
I agree.

And default values are already causing troubles, such as in https://github.com/flutter/flutter/pull/37896/files/daadb7b58e71117c5df325b9e1db70505f72beea#diff-96bb73b252a2e7c4b88089358c1ffd35R709 where I want to test the default value while reusing the test widget.

Reply:
in general we prefer to update the defaults everywhere and to be careful. defaults are self-documenting, and more idiomatic. We should use tests to verify that we haven't missed cases when we change them.